From 9e12b50d1886d09ea91964c1bc5805afeec16744 Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Tue, 30 Sep 2014 10:50:27 +0100
Subject: [PATCH] RHEL 7: v2v: Disable the --qemu-boot option (RHBZ#1147313).

This cannot work because there is no Gtk or SDL output mode
in RHEL 7's qemu-kvm.

In addition you will have to edit the -display option in the
qemu script.
---
 v2v/cmdline.ml                |  3 ++-
 v2v/virt-v2v-output-local.pod |  6 ++----
 v2v/virt-v2v.pod              | 13 -------------
 3 files changed, 4 insertions(+), 18 deletions(-)

diff --git a/v2v/cmdline.ml b/v2v/cmdline.ml
index 46f6910d0..aa89428c5 100644
--- a/v2v/cmdline.ml
+++ b/v2v/cmdline.ml
@@ -240,7 +240,6 @@ let parse_cmdline () =
                                     s_"Estimate size of source and stop";
     [ L"print-source" ], Getopt.Set print_source,
                                     s_"Print source and stop";
-    [ L"qemu-boot" ], Getopt.Set qemu_boot, s_"Boot in qemu (-o qemu only)";
     [ L"root" ],     Getopt.String ("ask|... ", set_root_choice),
                                     s_"How to choose root filesystem";
     [ L"vddk-config" ], Getopt.String ("filename", set_input_option_compat "vddk-config"),
@@ -588,6 +587,8 @@ read the man page virt-v2v(1).
         | Some d when not (is_directory d) ->
            error (f_"-os %s: output directory does not exist or is not a directory") d
         | Some d -> d in
+      if qemu_boot then
+        error (f_"-o qemu: the --qemu-boot option cannot be used in RHEL");
       Output_qemu.output_qemu os qemu_boot,
       output_format, output_alloc
 
diff --git a/v2v/virt-v2v-output-local.pod b/v2v/virt-v2v-output-local.pod
index 7427b1ed7..455d51fba 100644
--- a/v2v/virt-v2v-output-local.pod
+++ b/v2v/virt-v2v-output-local.pod
@@ -9,7 +9,7 @@ or libvirt
 
  virt-v2v [-i* options] -o local -os DIRECTORY
 
- virt-v2v [-i* options] -o qemu -os DIRECTORY [--qemu-boot]
+ virt-v2v [-i* options] -o qemu -os DIRECTORY
 
  virt-v2v [-i* options] -o null
 
@@ -47,12 +47,10 @@ where C<NAME> is the guest name.
 
 =item B<-o qemu -os> C<DIRECTORY>
 
-=item B<-o qemu -os> C<DIRECTORY> B<--qemu-boot>
-
 This converts the guest to files in C<DIRECTORY>.  Unlike I<-o local>
 above, a shell script is created which contains the raw qemu command
 you would need to boot the guest.  However the shell script is not
-run, I<unless> you also add the I<--qemu-boot> option.
+run.
 
 =item B<-o null>
 
diff --git a/v2v/virt-v2v.pod b/v2v/virt-v2v.pod
index cf9464834..0e9627829 100644
--- a/v2v/virt-v2v.pod
+++ b/v2v/virt-v2v.pod
@@ -142,11 +142,6 @@ Since F<guest-domain.xml> contains the path(s) to the guest disk
 image(s) you do not need to specify the name of the disk image on the
 command line.
 
-To convert a local disk image and immediately boot it in local
-qemu, do:
-
- virt-v2v -i disk disk.img -o qemu -os /var/tmp --qemu-boot
-
 =head1 OPTIONS
 
 =over 4
@@ -480,9 +475,6 @@ This is similar to I<-o local>, except that a shell script is written
 which you can use to boot the guest in qemu.  The converted disks and
 shell script are written to the directory specified by I<-os>.
 
-When using this output mode, you can also specify the I<--qemu-boot>
-option which boots the guest under qemu immediately.
-
 =item B<-o> B<rhev>
 
 This is the same as I<-o rhv>.
@@ -758,11 +750,6 @@ Print information about the source guest and stop.  This option is
 useful when you are setting up network and bridge maps.
 See L</Networks and bridges>.
 
-=item B<--qemu-boot>
-
-When using I<-o qemu> only, this boots the guest immediately after
-virt-v2v finishes.
-
 =item B<-q>
 
 =item B<--quiet>
-- 
2.21.0

